GeometricSeqPhenotype.java

```diff
@@ -0,0 +1,308 @@
+ *
+ * @author Thien Tran
+ */
+public class GeometricSeqPhenotype extends GeometricPhenotype {
+    /**
+     * The nucleotide sequence of this GeometricSeqPhenotype
+     */
+    private char[] nucleotideSequence;
+
+    /**
+     * The first parameter that determines the position of this GeometricPhenotype in Euclidean space
+     */
+    private double traitA;
+
+    /**
+     * The second parameter that determines the position of this GeometricPhenotype in Euclidean space
+     */
+    private double traitB;
+
+    /**
+     * The number of epitope mutations this GeometricPhenotype went through (counting from the startingSequence)
+     */
+    private int eMutation = 0;
+
+    /**
+     * The number of non-epitope mutations this GeometricPhenotype went through (counting from the startingSequence)
+     */
+    private int neMutation = 0;
+
+    /**
+     * Run expensive tests iff DEBUG == true.
+     */
+    public static final boolean DEBUG = false;
+
+    // Abstraction Function:
+    // A GeometricSeqPhenotype, gsp, is null if gsp.sequence = null,
+    // otherwise gsp.sequence = sequence for sequence.length() > 0 and sequence.length() % 3 == 0.
+    //
+    // The location of gsp in Euclidean space is defined by (traitA, traitB) each gsp is associated with a sequence of nucleotides.
+    // gsp1.sequence and gsp2.sequence are allowed to be equal for two given GeometricSeqPhenotype
+
+    // Representation invariant for every GeometricSeqPhenotype:
+    // gsp.sequence != null && gsp.sequence.length() > 0 && gsp.sequence.length() % 3 == 0 && gsp.sequence.length() == startingSequence.length()
+    // for all indexSite such that gsp.sequence.charAt(i):
+    //     indexSite is a character in ALPHABET
+    // for all codon such that gsp.sequence.charAt(i) + gsp.sequence.charAt(i + 1) + gsp.sequence.charAt(i + 2):
+    //     codon is not "STOP"
+
+    /**
+     * Constructor that creates a new GeometricSeqPhenotype.
+     *
+     * @effects Constructs a new GeometricSeqPhenotype that has a random starting sequence of length Parameters.sequence.length()
+     * and an empty GeometricPhenotype
+     */
+    public GeometricSeqPhenotype() {
+        this.nucleotideSequence = new char[Parameters.startingSequence.length()];
+        startingSequenceGenerator();
+        checkRep();
+    }
+
+    /**
+     * Constructor that creates a new GeometricSeqPhenotype.
+     *
+     * @param tA
+     * @param tB
+     * @effects Constructs a new GeometricSeqPhenotype that has a random starting sequence of length Parameters.sequence.length()
+     * and GeometricPhenotype with the data content of the given parameters.
+     */
+    public GeometricSeqPhenotype(double tA, double tB) {
+        traitA = tA;
+        traitB = tB;
+
+        startingSequenceGenerator();
+        checkRep();
+    }
+
+    // Generates a random sequence of nucleotides of length Parameters.startingSequence.length()
```

Comment:
Maybe describe when this would be used? Is it when the user doesn't have a specific sequence to input, and instead want to use a random sequence?